Userland/Libraries/LibVT/Terminal.cpp

```diff
@@ -1089,7 +1119,38 @@ void Terminal::execute_escape_sequence(Intermediates intermediates, bool ignore,
             }
             return;
         }
+        break;
+    case '(':
+    case ')':
+    case '*':
+    case '+':
+        // Determine G0..G3 index
+        int working_set_index = intermediate - '(';
+
+        CharacterSet new_set;
+        switch (last_byte) {
+        case 'B':
+            new_set = CharacterSet::Iso_8859_1;
+            break;
+        case '0':
+            new_set = CharacterSet::VT100;
+            break;
+        case 'U':
+            new_set = CharacterSet::Null;
+            break;
+        case 'K':
+            new_set = CharacterSet::UserDefined;
+            break;
+        default:
+            unimplemented_escape_sequence(intermediates, last_byte);
+            return;
+        }
+
+        dbgln_if(TERMINAL_DEBUG, "Setting G{} working set to character set {}", working_set_index, to_underlying(new_set));
+        m_working_sets[working_set_index] = new_set;
```

Comment:
Should this bounds-check `working_set_index`?

Reply:
Ah, I see. Maybe add a `VERIFY()` to make this more obvious, but I suppose it's pretty obvious already :)

Reply:
`VERIFY()` is now in!